cmd/manager/profileparser.go

```diff
@@ -113,6 +115,39 @@ func updateProfileBundleStatus(pcfg *profileparser.ParserConfig, pb *cmpv1alpha1
 	}
 }
 
+// fetechProfileRuleMap fetches all the profiles and use names as keys and
+// rules as values in a list
+func fetchProfileRuleMap(pcfg *profileparser.ParserConfig) (map[string][]cmpv1alpha1.ProfileRule, error) {
```

Comment:
A map that maps the other direction is likely more efficient.

For each profile, we go through its list of rules and add to a map where:
- the `rule.Name` as they key, and
- the value is a list of `profile.Name`

At the end, the map has all the rules which were selected in a profile, and the values are the profiles in which they were found.